test/specs/smtp_spec.rb

```diff
@@ -7,8 +7,8 @@
 BODY     = 'foo'
 
 class Jackal::Mail::Smtp
-  def send_mail(args)
-    $args = args
+  def send_mail(payload, args)
+    payload.set(:smtp_args, args)
```

Comment:
not even for the tests?... I experienced weirdness with global vars in the context of multiple tests before, so I thought it would just be cleaner to pass the test results to inspect in the actual payload...